pkg/synchromanager/clustersynchro/cluster_synchro.go

```diff
@@ -322,6 +323,15 @@ func (s *ClusterSynchro) setSyncResources() {
 	removedStorageGVRs := NewGVRSet()
 	s.storageResourceSynchros.Range(func(key, _ interface{}) bool {
 		storageGVR := key.(schema.GroupVersionResource)
+		for _, groupResources := range syncResources {
+			if hasString(groupResources.Resources, storageGVR.Resource) {
+				if groupResources.Versions != nil && !hasString(groupResources.Versions, storageGVR.Version) {
+					return true
+				}
+				delete(deleted, storageGVR)
+				return true
+			}
+		}
```

Comment:
It does not seem to have much to do with this pr, you can create a new pr and describe it.
This allows the pr to stay focused, making it easier to review and merge.

Reply:
delete(deleted, storageGVR) will keep the ResourceSynchro in storageResourceSynchros. genClusterStatus generates the ResourceSynchro status by storageResourceSynchros.
if we removed ResourceSynchro in storageResourceSynchros, the error message reported by reflector cann't be displayed.